internal/internal_test.go

```diff
@@ -251,73 +252,303 @@ func TestCheckWithLoggerError(t *testing.T) {
+	var result int32
+	result, err = getResponseStatus(input)
+
+	if err != nil {
+		t.Fatalf("Expected no error but got %v", err)
+	}
+
+	if result != int32(google_rpc.OK) {
+		t.Fatalf("Expected result %v but got %v", int32(google_rpc.OK), result)
+	}
+}
+
+func TestGetResponeHeaders(t *testing.T) {
+	input := make(map[string]interface{})
+
+	result, err := getResponseHeaders(input)
+	if err != nil {
+		t.Fatalf("Expected no error but got %v", err)
+	}
+
+	if len(result) != 0 {
+		t.Fatal("Expected no headers")
+	}
+
+	badHeader := "test"
+	input["headers"] = badHeader
+
+	_, err = getResponseHeaders(input)
+	if err == nil {
+		t.Fatal("Expected error but got nil")
+	}
+
+	testHeaders := make(map[string]interface{})
+	testHeaders["foo"] = "bar"
+	input["headers"] = testHeaders
+
+	result, err = getResponseHeaders(input)
+	if err != nil {
+		t.Fatalf("Expected no error but got %v", err)
+	}
+
+	if len(result) != 1 {
+		t.Fatalf("Expected one header but got %v", len(result))
+	}
+
+	testHeaders["baz"] = 1
+
+	result, err = getResponseHeaders(input)
+	if err != nil {
+		t.Fatalf("Expected no error but got %v", err)
+	}
+
+	if len(result) != 2 {
+		t.Fatalf("Expected two headers but got %v", len(result))
+	}
+}
+
+func testAuthzServer(customLogger plugins.Plugin) *envoyExtAuthzGrpcServer {
+
+	// Define a RBAC policy to allow or deny requests based on user roles
+	module := `
+		package istio.authz
+
+		import input.attributes.request.http as http_request
+
+		default allow = false
+
+		allow {
+			roles_for_user[r]
+			required_roles[r]
+		}
+
+		allow {
+			input.parsed_path = ["my", "test", "path"]
+		}
+
+		roles_for_user[r] {
+			r := user_roles[user_name][_]
+		}
+
+		required_roles[r] {
+			perm := role_perms[r][_]
+			perm.method = http_request.method
+			perm.path = http_request.path
+		}
+
+		user_name = parsed {
+			[_, encoded] := split(http_request.headers.authorization, " ")
+			[parsed, _] := split(base64url.decode(encoded), ":")
+		}
+
+		user_roles = {
+			"alice": ["guest"],
+			"bob": ["admin"]
+		}
+
+		role_perms = {
+			"guest": [
+				{"method": "GET",  "path": "/productpage"},
+			],
+			"admin": [
+				{"method": "GET",  "path": "/productpage"},
+				{"method": "GET",  "path": "/api/v1/products"},
+			],
+		}`
+
+	m, err := getPluginManager(module, customLogger)
 	if err != nil {
 		panic(err)
 	}
 
-	m.Register("test_plugin", customLogger)
-	config, err := logs.ParseConfig([]byte(`{"plugin": "test_plugin"}`), nil, []string{"test_plugin"})
+	query := "data.istio.authz.allow"
+	parsedQuery, err := ast.ParseBody(query)
+	if err != nil {
+		panic(err)
+	}
 
-	plugin := logs.New(config, m)
-	m.Register(logs.Name, plugin)
+	s := &envoyExtAuthzGrpcServer{
+		cfg: Config{
+			Addr:        ":50052",
+			Query:       query,
+			parsedQuery: parsedQuery,
+		},
+		manager: m,
+	}
+	return s
+}
 
-	if err := m.Start(ctx); err != nil {
+func testAuthzServerwithObjectDecision(customLogger plugins.Plugin) *envoyExtAuthzGrpcServer {
```

Comment:
nit: camelCasing should be `testAuthzServerWithObjectDecision` (capital `With`)

Reply:
Done.